webview/src/plots/components/vegaLite/util.ts

```diff
@@ -0,0 +1,181 @@
+import type { TopLevelSpec } from 'vega-lite'
+import {
+  AnchorDefinitions,
+  PLOT_COLOR_ANCHOR,
+  PLOT_HEIGHT_ANCHOR,
+  PLOT_WIDTH_ANCHOR,
+  PLOT_SHAPE_ANCHOR,
+  PLOT_STROKE_DASH_ANCHOR,
+  PLOT_TITLE_ANCHOR,
+  PLOT_X_LABEL_ANCHOR,
+  PLOT_Y_LABEL_ANCHOR,
+  PLOT_ZOOM_AND_PAN_ANCHOR,
+  ZOOM_AND_PAN_PROP
+} from 'dvc/src/cli/dvc/contract'
+import { DEFAULT_NB_ITEMS_PER_ROW } from 'dvc/src/plots/webview/contract'
+import { isMultiViewPlot } from 'dvc/src/plots/vega/util'
+import { truncate } from 'vega-util'
+
+const isAnchor = (
+  value: unknown,
+  anchorDefinitions: AnchorDefinitions
+): value is keyof AnchorDefinitions =>
+  typeof value === 'string' && value in anchorDefinitions
+
+const replaceAnchors = (
+  value: unknown,
+  anchorDefinitions: AnchorDefinitions
+  // eslint-disable-next-line sonarjs/cognitive-complexity
+): unknown => {
+  if (!value) {
```

Comment:
Do we want to return early if `value === 0`?

Reply:
Since it is also tested on line 38 and value is returned in the end as well, we should remove that check and let `isAnchor` return false if there is no value.